### PR TITLE
refactor: testsupport package segregation and cleanup

### PR DESCRIPTION
**Pull request by mfrancisc**
This PR contains: 

1. tries to addresses this discussion https://github.com/codeready-toolchain/toolchain-e2e/pull/779#discussion_r1297101930
2. created `testsupport/space` and `testsupport/spacebinding` packages and added the required imports where needed. I've tried to use dot import in order to limit the number of changes and thus limit the scope of the PR
3. create `testsupport/util` package in order to avoid import cycles
4. for the time being both `testsupport/space/spacerequest.go` and `testsupport/spacebinding/spacebindingrequest.go` are using `WithNamespace` but we can change it to `InNamespace` of course
5. delete some unused files and functions

NOTEs: 
- where `testsupport/space` and `testsupport/spacebinding` packages are imported I couldn't use dot import for both since the `WithNamespace` option is present in both
- I've tried to reduce the amount of changes to be review , and that's why I haven't moved the other files from `testsupport` package to new subpackages  like `testsupport/usersignup` , `testsupport/regsvc` and maybe others ... We might want to do this later.

### CODE REVIEW

**Review by rajivnathan — COMMENTED** [review]
Nice! Thanks for organizing the PRs to be easier to review 🙌  Just have a couple suggestions about package and file naming but otherwise looks great 👍

File: testsupport/util/util.go [hunk 1]
```diff
@@ -1,19 +1,11 @@
-package testsupport
+package util
```
[the remark this answers is not in the document]
Comment (mfrancisc) [reply testsupport/util/util.go]: Good point, renamed in https://github.com/codeready-toolchain/toolchain-e2e/pull/784/commits/875e3cc3ac292dcafa6422532b5b5f918db363ce

File: test/e2e/parallel/space_test.go [hunk 2]
```diff
@@ -8,6 +8,8 @@ import (
 	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
 	testspace "github.com/codeready-toolchain/toolchain-common/pkg/test/space"
 	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport/space"
+	testsupportsb "github.com/codeready-toolchain/toolchain-e2e/testsupport/spacebinding"
```
[the remark this answers is not in the document]
Comment (mfrancisc) [reply test/e2e/parallel/space_test.go:12]: Makes sense, let's be consistent, I've renamed `tsspace` to `testsupportspace` as you suggested, plz see https://github.com/codeready-toolchain/toolchain-e2e/pull/784/commits/875e3cc3ac292dcafa6422532b5b5f918db363ce

**Review by rajivnathan — APPROVED** [review]
Nice, thanks for the changes!

**Pull request merged**